Source/OxyPlot/Series/DataRange.cs

```diff
@@ -0,0 +1,119 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="DataRange.cs" company="OxyPlot">
+//   Copyright (c) 2014 OxyPlot contributors
+// </copyright>
+// <summary>
+//   Represents an interval defined by two doubles.
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace OxyPlot.Series
+{
+    using System;
+
+    /// <summary>
+    /// Represents an interval defined by two doubles.
+    /// </summary>
+    public struct DataRange : ICodeGenerating
+    {
+        /// <summary>
+        /// The undefined data range.
+        /// </summary>
+        public static readonly DataRange Undefined = new DataRange();
+
+        private readonly double minimum;
+        private readonly double maximum;
+        private readonly bool isDefined;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DataRange"/> struct.
+        /// </summary>
+        /// <param name="min">The inclusive lower bound.</param>
+        /// <param name="max">The inclusive upper bound.</param>
+        public DataRange(double min, double max)
+        {
+            if (double.IsNaN(min) || double.IsNaN(max))
+            {
+                throw new ArgumentException("NaN values are not permitted");
+            }
+
+            if (max < min)
+            {
+                throw new ArgumentException("max must be larger or equal min");
+            }
+
+            this.minimum = min;
+            this.maximum = max;
+
+            this.isDefined = true;
+        }
+
+        /// <summary>
+        /// Gets the lower bound of the data range.
+        /// </summary>
+        public double Minimum => this.minimum;
+
+        /// <summary>
+        /// Gets the upper bound of the data range.
+        /// </summary>
+        public double Maximum => this.maximum;
+
+        /// <summary>
+        /// Gets the difference between maximum and minimum.
+        /// </summary>
+        public double Range => this.Maximum - this.Minimum;
```

Comment:
Maybe not the best name... but I can't think of a better one.

Reply:
Measurement studio has a similar class named `Range` where the corresponding property is named 'Interval'. Another alternative might be 'Span'. I personally don't have a strong opinion about which name is best.